src/test/java/org/sonar/plugins/stash/StashIssueReportingPostJobTest.java

```diff
@@ -247,6 +252,36 @@ public void testExecuteOnWithNoDiffReport() throws Exception {
     verify(stashRequestFacade, times(0)).resetPullRequestApproval(eq(pr), (StashClient)Mockito.anyObject());
   }
 
+  @Test
+  public void testShouldApprovePullRequest() {
+    Issue minorIssue = new DefaultIssue().setSeverity(Severity.MINOR);
+    Issue majorIssue = new DefaultIssue().setSeverity(Severity.MAJOR);
+
+    report = new ArrayList<>();
+
+    report.add(minorIssue);
+    report.add(majorIssue);
+
+    assertFalse(
+        StashIssueReportingPostJob.shouldApprovePullRequest(Optional.empty(), report)
+    );
+
+    report.clear();

```

Comment:
I totally agree with you if the object under test is complex and contains internal state. However for a static method I think it is overkill. I very much prefer a simple test execution where I can see the difference of the parameters between the assertions very easily.

It would be even better to have nice parameterized tests in Java like [in python](https://docs.pytest.org/en/latest/parametrize.html#parametrize), but in Junit this involves a ridiculous amount of boilerplate hiding the actual logic.

Can we agree to disagree on this one with me (ab)using my maintainer powers to keep the status quo?